assignment4/assignment4.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <math.h>
+#include <vector>
+
+class IslandCounter{
+	void collapse_island(std::vector< std::vector<bool> > &map, int i, int j){
+		if(!map[i][j]) return;
+
+		map[i][j] = false;
+		collapse_island( map, std::min(i+1, (int) (map.size()-1)), j);
+		collapse_island( map, std::max(i-1, 0), j);
+		collapse_island( map, i, std::min(j+1,(int) (map[i].size()-1)));
+		collapse_island( map, i, std::max(j-1,0));
```

Comment:
Still need a stack, but on the heap you can usually get gigabytes of ram, while THE stack is by default a few megabytes. Don't worry about the other algorithm, you already got the new assignment :)